server/src/main/java/org/opensearch/identity/internal/InternalAuthenticationManager.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.internal;
+
+import org.opensearch.authn.realm.InternalRealm;
+import org.opensearch.identity.AccessTokenManager;
+import org.opensearch.identity.AuthenticationManager;
+import org.opensearch.authn.Subject;
+import org.apache.shiro.SecurityUtils;
+import org.apache.shiro.mgt.DefaultSecurityManager;
+import org.apache.shiro.mgt.SecurityManager;
+
+/**
+ * Implementation of authentication manager that does not enforce authentication
+ *
+ * This class and related classes in this package will not return nulls or fail permissions checks
+ *
+ * @opensearch.internal
+ */
+public class InternalAuthenticationManager implements AuthenticationManager {
```

Comment:
Maye add a note or two about what this class does? I am personally not sure what the security manager does after it is created here.